File: python/MyClass.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys, os

class MyClass:
    def __init__(self, id, name):
        self.id = id
        self.name = name

    def display(self):
        print "id: " + str(self.id) + ", name: " + self.name

def call_method():
    m = MyClass(1, "hoge")
    m.display()

def process():
    call_method()

if __name__ == "__main__":
    process()

File: ruby/my_class.rb
#!/usr/bin/env ruby
# coding: utf-8

class MyClass
  @val  = 1 # instance value
  @@val = 2 # class value
  
  # instance method
  def hoge
    puts 'hoge'
  end

  # class method
  class << self
    def _hoge
      puts '_hoge'
    end

    def _fuga
      puts '_fuga'
    end
  end
  
  def self._piyo
    puts '_piyo'
  end
end

a = MyClass.new
a.hoge

MyClass._hoge
MyClass._fuga
